W-17402451-SFTP-Connector-Override-Standard-Write-Strategy-KS (#2756)
mulesoft/docs-connectors: sftp/2.4/modules/ROOT/pages/sftp-connector-upgrade-guide.adoc

This release contains the following changes:

* Support for the Preferred Authentication Method `KEYBOARD_INTERACTIVE` is removed, as it isn't in use.

> * The connector has a new custom xref:connectors::sftp/sftp-write.adoc#write-strategy[Write strategy] field that enables you to configure the strategy to write a file.

Reviewer: ```suggestion
* The connector has a new custom xref:connectors::sftp/sftp-write.adoc#write-strategy[*write strategy*] field that enables you to configure the strategy to write a file.
```

Reviewer: I'm suggesting this change because:
* I assume the field name should not be translated, so it should be between asterics (*).
* As the "Use Write Strategy" title isn’t appropriate in the sentence, the hardcoded text is writen as in-line text.

If you face SSH connection issues due to unsupported ciphers, macs, hostKeyAlgorithms or kexAlgorithms, then change the priority or enable deprecated ciphers and algorithms forcefully. Although using this method may introduce additional risk. Refer to xref:sftp-security-configuration-override.adoc[Configure Security Parameters of SSH Connection] and xref:sftp-documentation.adoc[SFTP Connector Reference].

=== Override Standard Write Strategy

> If your file write operation overrides instead of appending when using the `APPEND` mode, use the `CUSTOM` write strategy as it is not applicable to the `OVERWRITE` or `CREATE_NEW` modes. This strategy enables you to manually specify the offset for the server to write the file in `APPEND` mode, particularly only for older SFTP servers that do not directly support write modes. Use this strategy only if the `STANDARD` strategy results in errors during file writing in `APPEND` mode and does not offer concurrency. Additionally, you can adjust the buffer size in the *Buffer Size for Write Strategy* field. 

Reviewer: ```suggestion
If your *File Write* operation overrides instead of appending when using the `APPEND` mode, use the `CUSTOM` write strategy as it isn’t applicable to the `OVERWRITE` or `CREATE_NEW` modes. With this strategy, you can manually specify the offset for the server to write the file in `APPEND` mode, particularly only for older SFTP servers that don’t directly support write modes. Use this strategy only if the `STANDARD` strategy results in errors during file writing in `APPEND` mode and doesn’t offer concurrency. Additionally, you can adjust the buffer size in the *Buffer Size for Write Strategy* field. 
```

Reviewer: Please, review:

- Capitalization: https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#:~:text=Except%20in%20titles%20and%20headings%2C%20do%20not%20capitalize%20common%20nouns.
- Font choices: https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#:~:text=full%20file%20name.-,font%20choices,-Follow%20these%20rules
Thanks!
